File: bin/word_count.py
# Input and output file paths
input_file = "hdfs://localhost:9000/user/root/input/words_small.txt"
output_file = "hdfs://localhost:9000/user/root/output/wordCount.csv"

# Read the input text file into an RDD
text_rdd = sc.textFile(input_file)

# Split each line into words and flatten the result
words_rdd = text_rdd.flatMap(lambda line: line.split())

# Count occurrences of each word
word_count_rdd = words_rdd.map(lambda word: (word, 1)).reduceByKey(lambda a, b: a + b)

# Save the result to the output file
word_count_rdd.saveAsTextFile(output_file)

File: bin/wordCount.scala
import org.apache.spark.rdd.lineage.LineageContext
import org.apache.spark.rdd.lineage.Conversions._

val input_file: String = "hdfs://localhost:9000/user/root/input/words_small.txt"
val output_file: String = "hdfs://localhost:9000/user/root/output/wordCount.csv"

val lc = new LineageContext(sc)
val data=lc.textFile(input_file)

val splitdata = data.flatMap(l => l.split(" "))
val mapdata = splitdata.map(w => (w,1))
val reducedata = mapdata.reduceByKey(_+_)

reducedata.saveAsTextFile(output_file)